02/studenttests.ml

```diff
@@ -481,19 +518,64 @@ let instruction_tests = [
       && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = Int64.sub Int64.max_int 254L
     )
   );
+  ("setb_cmp", Gradedtests.machine_test "1" 6 (set_cmp 10L 10L Eq)
+    (fun m ->
+      let exp = Int64.sub Int64.max_int 254L in
+        m.regs.(rind Rax) = exp
+        && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = exp
+    )
+  );
+  ("setb_cmp", Gradedtests.machine_test "1" 6 (set_cmp 11L 10L Eq)
+    (fun m ->
+      let exp = Int64.sub Int64.max_int 255L in
+        m.regs.(rind Rax) = exp
+        && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = exp
+    )
+  );
+  ("setb_cmp", Gradedtests.machine_test "1" 6 (set_cmp 10L 10L Ge)
+    (fun m ->
+      let exp = Int64.sub Int64.max_int 254L in
+        m.regs.(rind Rax) = exp
+        && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = exp
+    )
+  );
+  ("setb_cmp", Gradedtests.machine_test "1" 6 (set_cmp (-10L) 10L Le)
+    (fun m ->
+      let exp = Int64.sub Int64.max_int 254L in
+        m.regs.(rind Rax) = exp
+        && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = exp
+    )
+  );
   ("addq", Gradedtests.machine_test "rax=24L" 3 (bin_aop Addq)
     (fun m -> m.regs.(rind Rax) = 24L));
   ("subq", Gradedtests.machine_test "rax=24L" 3 (bin_aop Subq)
     (fun m -> m.regs.(rind Rax) = -20L));
+  ("leaq", Gradedtests.machine_test "90()" 2 (leaq_simpl 0L (Ind1 (Lit 90L)))
+    (fun m -> m.regs.(rind Rax) = 90L));
+  ("leaq", Gradedtests.machine_test "$90" 2 (leaq_simpl 10L (Ind2 Rax))
+    (fun m -> m.regs.(rind Rax) = 10L));
+  ("leaq", Gradedtests.machine_test "$90" 2 (leaq_simpl 10L (Ind3 (Lit 10L, Rax)))
+    (fun m -> m.regs.(rind Rax) = 20L));
 ]
 
+let redefinedsym_test (p:prog) () =
+  try ignore (assemble p);
+    failwith "Should have raised Redefined_sym"
+  with
+    | Redefined_sym _ -> ()
+    | _ -> failwith "Should have raised Redefined_sym"
+
 let provided_tests : suite = [
   Test ("Debug: End-to-end Tests", [
     ("empty main", Gradedtests.program_test [empty_main] 0L)
     ; ("negq", Gradedtests.program_test (main_driver::negq) (Int64.neg 99L) )
     ; ("addq", Gradedtests.program_test (main_driver::addq) 3L )
     ; ("fibonacci", Gradedtests.program_test (main_driver::fibonacci) 832040L)
   ]);
+  Test ("Debug: Condition Flags Set Tests", [
+    ("redefined sym", redefinedsym_test [text "main" [Retq,[]]; text "main" [Retq,[]]])
```

Comment:
Probably those should go into e2e or create a new assemble category since the errore is thrown there